tests/unit/labels/LabelsEncoder_unittest.cc

```diff
@@ -0,0 +1,169 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/labels/BinaryLabelEncoder.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/labels/MulticlassLabels.h>
+#include <shogun/labels/MulticlassLabelsEncoder.h>
+using namespace shogun;
+
+TEST(BinaryLabelEncoder, fit_transform)
+{
+	auto label_encoder = std::make_shared<BinaryLabelEncoder>();
+	SGVector<int32_t> vec{-1, -1, 1, -1, 1};
+	auto origin_labels = std::make_shared<BinaryLabels>(vec);
+	auto unique_vec = label_encoder->fit(origin_labels);
+	EXPECT_EQ(-1, unique_vec[0]);
+	EXPECT_EQ(1, unique_vec[1]);
+
+	auto result_labels = label_encoder->transform(origin_labels);
+	auto result_vec = result_labels->as<BinaryLabels>()->get_labels();
+	EXPECT_EQ(-1, result_vec[0]);
+	EXPECT_EQ(-1, result_vec[1]);
+	EXPECT_EQ(1, result_vec[2]);
+	EXPECT_EQ(-1, result_vec[3]);
+	EXPECT_EQ(1, result_vec[4]);
+
+	auto inv_result = label_encoder->inverse_transform(result_labels)
+	                      ->as<BinaryLabels>()
+	                      ->get_labels();
+
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(vec[i], inv_result[i]);
+	}
+}
+
+TEST(BinaryLabelEncoder, labels_not_neg1_or_1)
+{
+	auto label_encoder = std::make_shared<BinaryLabelEncoder>();
+	SGVector<int32_t> vec{-100, 200, -100, 200, -100};
+	auto origin_labels = std::make_shared<BinaryLabels>(vec);
+	auto unique_vec = label_encoder->fit(origin_labels);
+	EXPECT_EQ(-100, unique_vec[0]);
+	EXPECT_EQ(200, unique_vec[1]);
+
+	auto result_labels = label_encoder->transform(origin_labels);
+	auto result_vec = result_labels->as<BinaryLabels>()->get_labels();
+	EXPECT_EQ(-1, result_vec[0]);
+	EXPECT_EQ(1, result_vec[1]);
+	EXPECT_EQ(-1, result_vec[2]);
+	EXPECT_EQ(1, result_vec[3]);
+	EXPECT_EQ(-1, result_vec[4]);
+
+	auto inv_result = label_encoder->inverse_transform(result_labels)
+	                      ->as<DenseLabels>()
+	                      ->get_labels();
+
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(vec[i], inv_result[i]);
+	}
+
+	SGVector<int32_t> test_vec{-1, -1, -1, -1, -1, 1};
+	auto test_labels = std::make_shared<BinaryLabels>(test_vec);
+	auto inv_test = label_encoder->inverse_transform(test_labels)
+	                    ->as<BinaryLabels>()
+	                    ->get_labels();
+	EXPECT_EQ(-100, inv_test[0]);
```

Comment:
could you use a loop (or macro) and the original vector so make this less verbose?